seasmon_xr/accessors.py

```diff
@@ -642,3 +646,16 @@ def mean(
         return xarray.DataArray(
             data=data, dims=dims, coords=coords, attrs={}, name=name
         )
+
+
+@xarray.register_dataset_accessor("hdc")
+@xarray.register_dataarray_accessor("hdc")
+class HDC:
+    """xarray accessor for HDC xarray tools."""
+
+    def __init__(self, xarray_obj):
+        self.algo = PixelAlgorithms(xarray_obj)
+        self.anom = Anomalies(xarray_obj)
+        self.iteragg = IterativeAggregation(xarray_obj)
+        self.whit = WhittakerSmoother(xarray_obj)
+        self.zonal = ZonalStatistics(xarray_obj)
```

Comment:
so this will make it impossible to construct hdc on a Dataset since Zonal thing can't do it.

We will make it lazy at some point, but I guess these are low cost at construction time so maybe not a big deal.

This is fine for now, just need to make sure this works on Dataset objects for those sub-accessors.

Reply:
oops sorry, it fails at call site, ok then

Reply:
Why wouldn't it work? There's only an `NotImplementedError` raised if you call `xx.hdc.zonal.mean` with `xx` being a dataset.

I was thinking of either adding an argument to `mean` that let's you choose the band, or just handle multiple bands

Reply:
see comment above, diff makes it look like raise is in constructor, but it's not so this is fine. Constructing those accessors lazily would be nice, but it's tricky to make it work well with typing, so unless some of those are expensive at construction time this approach should be fine.

Reply:
ok, should we merge this and then do another PR removing the grouped accessors once they are gone from `seasmon-aws`?

Reply:
yes, but please use squash option, this is really just one commit worth of change in here